configure.ac

```diff
@@ -498,5 +498,9 @@ fi
 AC_SUBST([BASH_COMPLETION_DIR])
 AM_CONDITIONAL([INSTALL_BASH_COMPLETION],[test "x$with_bash_completion_dir" != "xno"])
 
+if test "x$ac_cv_func_clock_gettime" != "xyes" ||
+    test "x$ac_cv_lib_rt_clock_gettime" != "xyes"; then
```

Comment:
Is `ac_cv_lib_rt_clock_gettime` set to `yes` even if the function is exposed via libc?

I can completely believe that it would be, but I just want to double-check that autoconf is insane in the way you're assuming here.

Reply:
Looks like it.  `configure` contains:

```
# Check for clock_gettime.  On some systems, this is provided via librt.
{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for clock_gettime in -lrt" >&5
$as_echo_n "checking for clock_gettime in -lrt... " >&6; }
if ${ac_cv_lib_rt_clock_gettime+:} false; then :
  $as_echo_n "(cached) " >&6
else
  ac_check_lib_save_LIBS=$LIBS
LIBS="-lrt  $LIBS"
cat confdefs.h - <<_ACEOF >conftest.$ac_ext
/* end confdefs.h.  */

/* Override any GCC internal prototype to avoid an error.
   Use char because int might match the return type of a GCC
   builtin and then its argument prototype would still apply.  */
#ifdef __cplusplus
extern "C"
#endif
char clock_gettime ();
int
main ()
{
return clock_gettime ();
  ;
  return 0;
}
_ACEOF
if ac_fn_c_try_link "$LINENO"; then :
  ac_cv_lib_rt_clock_gettime=yes
else
  ac_cv_lib_rt_clock_gettime=no
fi
{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_rt_clock_gettime" >&5
$as_echo "$ac_cv_lib_rt_clock_gettime" >&6; }
if test "x$ac_cv_lib_rt_clock_gettime" = xyes; then :
  cat >>confdefs.h <<_ACEOF
#define HAVE_LIBRT 1
_ACEOF
```

I mean, assuming that adding a potentially-unnecessary `-lrt` to the linker won't break it.  Which seems to be an assumption that autotools makes.

Reply:
Hmm.  The case of "adding `-lrt` breaks things" is basically what our `POSIX/posix-l.sh` checks for... can you check if this breaks on OS X?  IIRC it now has clock_gettime in libc but does not have a librt?  In FreeBSD I added some special handling for libraries which POSIX says should exist but don't... I'm not sure if OS X handles that correctly.

It might be safer to tell autoconf "link to librt if we can" followed by "check if clock_gettime works" without specifying where to find clock_gettime.

Reply:
I did check that it works on OS X (https://github.com/Tarsnap/tarsnap/pull/287#issuecomment-334334811), with both the native headers and the XCode 8 headers.  But I'll poke around and look at why `POSIX/posix-l.sh` is needed for some things but apparently not with autotools.

Reply:
Oh, I see what the autoconf bits are doing.  They're doing exactly what I suggested: Checking "does linking to librt make clock_gettime work", and then checking "does clock_gettime work".

So I think the answer here is just to keep the `ac_cv_func_clock_gettime` check but drop the `ac_cv_lib_rt_clock_gettime` check -- the former tells us whether we have a working `clock_gettime`, and the latter just says where to find it (which we don't care about).

Reply:
BTW on the libraries issue, autoconf and libcperciva take opposing approaches: Autoconf says "check if we need library foo in order to get symbol bar; if yes, link to it", while the libcperciva code says "bar might be defined by library foo, and POSIX says that foo should exist, so link to it unless it's broken".  In other words, autoconf defaults to not linking to libraries unless they're determined to be needed, while libcperciva defaults to linking to libraries unless they're determined to be broken.